persistence-modules/spring-boot-persistence-2/src/test/java/com/baeldung/db/indexing/FileSystemImageIntegrationTest.java

```diff
@@ -0,0 +1,58 @@
+package com.baeldung.db.indexing;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.InputStream;
+
+import org.assertj.core.api.Assertions;
+import org.junit.FixMethodOrder;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMultipartHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import com.baeldung.db.indexing.ImageUploaderApplication;
+
+@SpringBootTest(classes = ImageUploaderApplication.class)
+@AutoConfigureMockMvc
+@FixMethodOrder
+class FileSystemImageIntegrationTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Test
+    @Order(1)
+    void givenJpegImage_whenUploadIt_shallReturnItsId() throws Exception {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        InputStream image = classLoader.getResourceAsStream("baeldung.jpeg");
+
+        MockMultipartHttpServletRequestBuilder multipartRequest = MockMvcRequestBuilders.multipart("/file-system/image")
+            .file(new MockMultipartFile("image", "baeldung", MediaType.TEXT_PLAIN_VALUE, image));
+
+        MvcResult result = mockMvc.perform(multipartRequest)
+            .andExpect(status().isOk())
+            .andReturn();
+
+        Assertions.assertThat(result.getResponse()
+            .getContentAsString())
+            .isEqualTo("1");
+    }
+
+    @Test
+    @Order(2)
+    void givenBaeldungImage_whenDownloadIt_shallReturnTheImage() throws Exception {
```

Comment:
You can see that this test is part of the other one as its `given` describes something that's not in its body.

Reply:
From my point of view, the given is something expected for the test run as expected. Usually an argument, but it may be a pre-condition too.
Anyway, it is a little different now.

Reply:
@GilvanOrnelas you're right that `given` means _I expect this precondition for a test_. However, in test code, it's usual that the `given` part of the test is within the body of the test method, setting up that precondition so that each test can be run independently. In your previous version, for example, it wasn't possible to only execute the second test, as it depended on the execution of the first.

I've seen this done before, sometimes for very good reasons. I even wrote a JUnit framework once to allow tests to express their dependencies, and for the test execution order to be determined from them, rather than have to be hard-coded using `Order`.

However, on the whole, it's well worth avoiding this pattern.

Reply:
Cool. I would love to see your "dependency JUnit tests".

TestNG has an interesting feature - dependsOnMethods. Defining dependencies is a little better than forcing an order. I wonder if your framework looks like it.